TrackingTools/DetLayers/interface/rangesIntersect.h

```diff
@@ -12,15 +12,15 @@
 
 template <typename Range>
 inline bool rangesIntersect(const Range& a, const Range& b) {
-  return !((a.first > b.second) | (b.first > a.second));
+  return !((a.first > b.second) || (b.first > a.second));
```

Comment:
it looks like a wack-a-mole
https://github.com/cms-sw/cmssw/pull/40611#discussion_r1089028261

Reply:
Is there anything to be done here?

Reply:
It looks like the bit operation here was intentional. Isn't there any way to instruct clang to not complain about these?